aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -49,68 +51,75 @@ namespace {
   }
 }
 
+void s_addmm_out_csr_sparse_dense_cuda_worker(int64_t nnz, int64_t m, int64_t n, int64_t k, Tensor& r_, const Scalar& beta, const Tensor& t, const Scalar& alpha, Tensor& crow_indices, Tensor& col_indices, Tensor& values, const Tensor& dense) {
+
+  // No half support, so we don't have to use CUDATypeConversion
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(
+      values.scalar_type(), "addmm_sparse_cuda", [&] {
+        scalar_t cast_beta = beta.to<scalar_t>();
+        scalar_t cast_alpha = alpha.to<scalar_t>();
+        Tensor r__;
+        if (cast_beta == scalar_t(0)) {
+          r_.zero_();
+        } else if (!is_same_tensor(t, r_)) {
+          r_.copy_(t);
+        }
+        if (nnz > 0) {
+          if(r_.stride(0) == 1 && r_.stride(1) == r_.size(0)) {
+            r__ = r_;
+          } else {
+            // TODO: how... strange
```

Comment:
Not sure what this TODO is about. What the code does below, is just ensuring that `r__` is Fortran contiguous.
Perhaps add a comment explaining why this is preferred?